plugins/common/functions

```diff
@@ -185,13 +185,13 @@ is_valid_app_name() {
   local APP="$1"
   [[ -z "$APP" ]] && dokku_log_fail "APP must not be null"
   if [[ "$APP" =~ ^[a-z].* ]] || [[ "$APP" =~ ^[0-9].* ]]; then
-    if [[ ! $APP =~ [A-Z] ]] && [[ ! $APP =~ [:] ]]; then
+    if [[ ! $APP =~ [A-Z] ]] && [[ ! $APP =~ [:] ]] && [[ ! $APP =~ [_] ]]; then
```

Comment:
fyi, this added check forces 193, which applies to existing apps during upgrade. Just had to restore my apps :(

Reply:
Damn sorry about that. I'll fix now in 0.22.0.